### ISSUE TITLE
Panic when running with custom fixtures specified

### ISSUE THREAD

**Issue by ruudk** [stripe/stripe-mock#22]
```
$ stripe-mock -version
0.2.0
```

When I run `stripe-mock` with `-fixtures ~/fixtures3.yaml -spec ~/spec3.yaml` the server starts. The fixtures3.yaml and spec3.yaml are from https://raw.githubusercontent.com/stripe/openapi/5f2f0e245c07c918cca185902aceffc64a0b1957/openapi/fixtures3.yaml and https://raw.githubusercontent.com/stripe/openapi/5f2f0e245c07c918cca185902aceffc64a0b1957/openapi/spec3.yaml

But when I do 

```
## Request (3)
curl -X "POST" "http://localhost:12111/v1/sources" \
     -H "Authorization: Bearer sk_test_123" \
     -H "Content-Type: application/x-www-form-urlencoded; charset=utf-8" \
     --data-urlencode "type=bitcoin"
```

the server crashes

```
stripe-mock -fixtures ~/fixtures3.yaml -spec ~/spec3.yaml
Routing to 107 path(s) and 186 endpoint(s) with 104 validator(s)
Listening on port 12111
Request: POST /v1/sources
2017/10/03 10:08:13 http: panic serving 127.0.0.1:51820: Responding to /v1/sources:
Dereferencing '#/components/schemas/source':
Using fixture 'source':
Schema is an object:
{
  "properties": {
    "amount": {
      "nullable": true,
      "type": "integer"
    },
    "client_secret": {
      "type": "string"
    },
    "code_verification": {
      "$ref": "#/components/schemas/source_code_verification_flow"
    },
    "created": {
      "type": "integer"
    },
    "currency": {
      "nullable": true,
      "type": "string"
    },
    "flow": {
      "type": "string"
    },
    "id": {
      "type": "string"
    },
    "livemode": {
      "type": "boolean"
    },
    "metadata": {
      "nullable": true,
      "type": "object"
    },
    "object": {
      "type": "string"
    },
    "owner": {
      "anyOf": [
        {
          "$ref": "#/components/schemas/source_owner"
        }
      ],
      "nullable": true
    },
    "receiver": {
      "$ref": "#/components/schemas/source_receiver_flow"
    },
    "redirect": {
      "$ref": "#/components/schemas/source_redirect_flow"
    },
    "statement_descriptor": {
      "nullable": true,
      "type": "string"
    },
    "status": {
      "type": "string"
    },
    "type": {
      "type": "string"
    },
    "usage": {
      "nullable": true,
      "type": "string"
    }
  },
  "required": [
    "client_secret",
    "created",
    "flow",
    "id",
    "livemode",
    "object",
    "status",
    "type"
  ],
  "type": "object",
  "x-expandableFields": [
    "code_verification",
    "owner",
    "receiver",
    "redirect"
  ],
  "x-resourceId": "source"
}
But example is:
map[owner:map[verified_email:<nil> verified_name:<nil> verified_phone:<nil> address:<nil> email:jenny.rosen@example.com name:<nil> phone:<nil> verified_address:<nil>] redirect:map[url: return_url: status:] statement_descriptor: usage:single_use code_verification:map[attempts_remaining:%!s(int=0) status:] id:src_1B3xSU2eZvKYlo2CYdL110Ll object:source type:bitcoin client_secret:src_client_secret_BQp65YD6eXk9NLfGKOwigCoK created:%!s(int=1234567890) flow:receiver amount:%!s(int=1000) receiver:map[address:test_1MBhWS3uv4ynCfQXF3xQjJkzFPukr4K56N amount_charged:%!s(int=0) amount_received:%!s(int=0) amount_returned:%!s(int=0) refund_attributes_method:email refund_attributes_status:missing] status:pending currency:usd livemode:%!s(bool=false) metadata:map[]]
goroutine 20 [running]:
net/http.(*conn).serve.func1(0xc420dce8c0)
	/usr/local/Cellar/go/1.9/libexec/src/net/http/server.go:1697 +0xd0
panic(0x134df00, 0xc420900960)
	/usr/local/Cellar/go/1.9/libexec/src/runtime/panic.go:491 +0x283
main.(*DataGenerator).generateInternal(0xc42087fb00, 0xc420ae4780, 0xc42027c365, 0xb, 0x0, 0xc420900160, 0xc420076660, 0x60, 0xc420a69860, 0xc420a69930, ...)
	/Users/brandur/Documents/go/src/github.com/stripe/stripe-mock/generator.go:151 +0x16af
main.(*DataGenerator).Generate(0xc42087fb00, 0xc420ae4780, 0xc42027c365, 0xb, 0x0, 0x1, 0x0, 0x0, 0x0)
	/Users/brandur/Documents/go/src/github.com/stripe/stripe-mock/generator.go:22 +0x120
main.(*StubServer).HandleRequest(0xc420b5d9c0, 0x159b240, 0xc4208fc540, 0xc4203e2000)
	/Users/brandur/Documents/go/src/github.com/stripe/stripe-mock/server.go:175 +0x55b
main.(*StubServer).HandleRequest-fm(0x159b240, 0xc4208fc540, 0xc4203e2000)
	/Users/brandur/Documents/go/src/github.com/stripe/stripe-mock/main.go:75 +0x48
net/http.HandlerFunc.ServeHTTP(0xc4208f9f50, 0x159b240, 0xc4208fc540, 0xc4203e2000)
	/usr/local/Cellar/go/1.9/libexec/src/net/http/server.go:1918 +0x44
net/http.(*ServeMux).ServeHTTP(0x15e7c60, 0x159b240, 0xc4208fc540, 0xc4203e2000)
	/usr/local/Cellar/go/1.9/libexec/src/net/http/server.go:2254 +0x130
net/http.serverHandler.ServeHTTP(0xc42046fd40, 0x159b240, 0xc4208fc540, 0xc4203e2000)
	/usr/local/Cellar/go/1.9/libexec/src/net/http/server.go:2619 +0xb4
net/http.(*conn).serve(0xc420dce8c0, 0x159b8c0, 0xc42084a980)
	/usr/local/Cellar/go/1.9/libexec/src/net/http/server.go:1801 +0x71d
created by net/http.(*Server).Serve
	/usr/local/Cellar/go/1.9/libexec/src/net/http/server.go:2720 +0x288
```

I didn't make any changes to the files, so I'm not sure what is going on.

When I remove the `-fixture path.yaml` and only run it with `-spec path.yaml` it works fine.

Any idea?

**Comment by ruudk** [comment]
Just checked, it works with JSON. So I'll use that.

**Comment by brandur-stripe** [comment]
@ruudk Thanks for logging! I'll take a look.

Just OOC, you saw that if you don't specify either `-fixture` or `-spec` it'll just use the default spec/fixtures that are bundled into the executable right? Unless you're trying to accomplish something specific, this might be the easier path for you.

**Comment by ruudk** [comment]
@brandur-stripe Yep, but i wanted to change some responses. Then I found out that it didn't work as expected.

**Comment by brandur-stripe** [comment]
Alright, it turns out that this one comes from https://github.com/go-yaml/yaml/issues/139. We have a fix coming in #23 to address the problem.

**Comment by brandur-stripe** [comment]
@ruudk A fix has been released in v0.2.1.

**Comment by ruudk** [comment]
Wow, :zap: super fast! Thanks!
